middleware/kubernetes/federation_test.go

```diff
@@ -0,0 +1,110 @@
+package kubernetes
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/coredns/coredns/middleware/etcd/msg"
+	"github.com/miekg/dns"
+	"k8s.io/client-go/1.5/pkg/api"
+)
+
+func testStripFederation(t *testing.T, k Kubernetes, input []string, expectedFed string, expectedSegs string) {
+	fed, segs := k.stripFederation(input)
+
+	if expectedSegs != strings.Join(segs, ".") {
+		t.Errorf("For '%v', expected segs result '%v'. Instead got result '%v'.", strings.Join(input, "."), expectedSegs, strings.Join(segs, "."))
+	}
+	if expectedFed != fed {
+		t.Errorf("For '%v', expected fed result '%v'. Instead got result '%v'.", strings.Join(input, "."), expectedFed, fed)
+	}
+}
+
+func TestStripFederation(t *testing.T) {
+	k := Kubernetes{Zones: []string{"inter.webs.test"}}
+	k.Federations = []Federation{{name: "fed", zone: "era.tion.com"}}
+
+	testStripFederation(t, k, []string{"service", "ns", "fed", "svc"}, "fed", "service.ns.svc")
+	testStripFederation(t, k, []string{"service", "ns", "foo", "svc"}, "", "service.ns.foo.svc")
+	testStripFederation(t, k, []string{"foo", "bar"}, "", "foo.bar")
+
+}
+
+type apiConnFedTest struct{}
+
+func (apiConnFedTest) Run() {
```

Comment:
nit: you put these 1-line function on 1 line, even without newlines between them.